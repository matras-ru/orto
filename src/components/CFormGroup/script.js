import { selfInstall } from '@/utils';

import ThemeClass from '@/themes/default/CFormGroup';

const { baseClass } = ThemeClass;

export default {
    name: 'FormGroup',

    functional: true,

    install(Vue, theme) {
        selfInstall(Vue, theme, this);
    },

    render(h, { children }) {
        const componentData = {
            class: baseClass
        };

        return h('div', componentData, children);
    }
};
