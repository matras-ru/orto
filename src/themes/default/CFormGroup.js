const FormGroup = {
    baseClass: 'mb-1-4'
};

export default FormGroup;
