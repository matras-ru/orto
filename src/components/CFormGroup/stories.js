import { storiesOf } from '@storybook/vue';

import CFormGroup from './';

import '@/assets/css/tailwind.css'; // TODO: create a single loading of global styles -> FRONTEND-498

const stories = storiesOf('CFormGroup', module);

stories.add('Form group', () => ({
    components: { CFormGroup },
    template: '<CFormGroup>Form group here</CFormGroup>'
}));
