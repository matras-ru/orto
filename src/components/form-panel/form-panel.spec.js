/* eslint-env jest */
import { mount } from '@vue/test-utils';
import CFormGroup from './form-panel';

describe('Form panel', () => {
    it('is rendered', () => {
        const wrapper = mount(CFormGroup);

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });

    it('default slot', () => {
        const wrapper = mount(CFormGroup, {
            slots: {
                default: 'foobar'
            }
        });

        expect(wrapper.text()).toEqual('foobar');
    });
});
