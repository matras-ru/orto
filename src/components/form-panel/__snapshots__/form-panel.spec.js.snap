// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form panel is rendered 1`] = `
<div class="mb-2-2">
  <div>
    <div></div>
  </div>
</div>
`;
